File: cameralibrary/src/main/java/com/cgfay/camera/PreviewEngine.java
package com.cgfay.camera;

import android.app.Activity;

import com.cgfay.camera.model.AspectRatio;

import java.lang.ref.WeakReference;

/**
 * 相机预览引擎
 */
public final class PreviewEngine {

    private WeakReference<Activity> mWeakActivity;

    private PreviewEngine(Activity activity) {
        mWeakActivity = new WeakReference<>(activity);
    }

    public static PreviewEngine from(Activity activity) {
        return new PreviewEngine(activity);
    }


    /**
     * 设置长宽比
     * @param ratio
     * @return
     */
    public PreviewBuilder setCameraRatio(AspectRatio ratio) {
        return new PreviewBuilder(this, ratio);
    }

    public Activity getActivity() {
        return mWeakActivity.get();
    }



}

File: cameralibrary/src/main/java/com/cgfay/camera/PreviewEngine.kt
package com.cgfay.camera

import android.app.Activity
import com.cgfay.camera.model.AspectRatio

/**
 * Entry point for launching camera preview.
 */
class PreviewEngine private constructor(activity: Activity) {

    private val weakActivity = java.lang.ref.WeakReference(activity)

    companion object {
        /**
         * Create a [PreviewEngine] from an [Activity] context.
         */
        @JvmStatic
        fun from(activity: Activity): PreviewEngine = PreviewEngine(activity)
    }

    /**
     * Configure the preview with a given [AspectRatio].
     */
    fun setCameraRatio(ratio: AspectRatio): PreviewBuilder = PreviewBuilder(this, ratio)

    internal fun getActivity(): Activity? = weakActivity.get()
}
